app/controllers/concerns/foreman_openscap/hosts_controller_extensions.rb

```diff
@@ -10,5 +16,43 @@ def process_hostgroup
       @openscap_proxy = @hostgroup.openscap_proxy
       super
     end
+
+    def update_multiple_openscap_proxy
+      if (id = params['smart_proxy']['id'])
+        smart_proxy = ::SmartProxy.find(id)
+        @hosts.each do |host|
+          host.openscap_proxy = smart_proxy
+          host.save!
+        end
+        notice _("Updated hosts: Assigned with OpenSCAP Proxy: %s") % smart_proxy.name
+        # We prefer to go back as this does not lose the current search
+        redirect_to hosts_path
+      else
+        error _('No OpenSCAP Proxy selected.')
+        redirect_to(update_multiple_openscap_proxy_hosts_path)
+      end
+    end
+
+    private
+
+    def find_multiple
```

Comment:
I do not think you need this method. This module extends `HostsController` and there already is [find_multiple](https://github.com/theforeman/foreman/blob/develop/app/controllers/hosts_controller.rb#L725)